contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1328,6 +1341,148 @@ pltsql_post_parse_analyze(ParseState *pstate, Query *query, JumbleState *jstate)
 	}
 }
 
+static bool
+checkForJsonAuto(Query *query)
+{
+	// Detect JSON AUTO
+	List* target = query->targetList;
+	bool isJsonAuto = false;
+	List* rtable;
+	RangeTblEntry* rte;
+	Query* subq;
+	List* subqRtable;
+	RangeTblEntry* subqRte;
+	RangeTblEntry* queryRte;
+	Alias *colnameAlias;
+	int numTables = 0;
+	int currTables = 0;
+	int currMax = 0;
+	forjson_table **tableInfoArr;
+	if(target) {
+		ListCell* lc = list_nth_cell(target, 0);
+		if(lc != NULL && ((Node*) lfirst(lc))->type == T_TargetEntry) {
+			TargetEntry* te = lfirst_node(TargetEntry, lc);
+			if(te && strcmp(te->resname, "json") == 0 && te->expr != NULL && ((Expr*) te->expr)->type == T_FuncExpr) {
+				List* args = ((FuncExpr*) te->expr)->args;
+				if(args != NULL && ((Node*) linitial(args))->type == T_Aggref) {
+					Aggref* agg = linitial_node(Aggref, args);
+					List* aggargs = agg->args;
+					if(aggargs != NULL && list_nth_cell(aggargs, 1) != NULL && ((Node*) lsecond(aggargs))->type == T_TargetEntry) {
+						TargetEntry* te2 = lsecond_node(TargetEntry, aggargs);
+						if(te2->expr != NULL && ((Expr*) te2->expr)->type == T_Const) {
+							Const* c = (Const*) te2->expr;
+							if(c->constvalue == 0)
+								isJsonAuto = true;
+						}
+					}
+				}
+			}
+		}
+	}
+	
+	if(!isJsonAuto)
+		return false;
+
+	// Modify query to be of the form "JSONAUTOALIAS.[nest_level].[table_alias]" 
+	rtable = (List*) query->rtable;
+	if(rtable != NULL) {
```

Comment:
What if it's == NULL, in such case we should throw exception ? If so , pls use assert

Reply:
Otherwise we'll have to write some else clause to cover those